lib/shared/common_markdown_body.dart

```diff
@@ -152,3 +162,122 @@ class CommonMarkdownBody extends StatelessWidget {
     );
   }
 }
+
+/// A Markdown Extension to handle spoiler tags on Lemmy. This extends the [md.BlockSyntax]
+/// to allow for multi-line parsing of text for a given spoiler tag.
+///
+/// It parses the following syntax for a spoiler:
+///
+/// ```
+/// ::: spoiler spoiler_title
+/// spoiler_body
+/// :::
+/// ```
+class SpoilerBlockSyntax extends md.BlockSyntax {
+  /// The pattern to match the end of a spoiler
+  static final RegExp _spoilerBlockEnd = RegExp(r'^(?:::)$');
+
+  /// The pattern to match the beginning of a spoiler
+  @override
+  RegExp get pattern => RegExp(r'^:::\s+spoiler\s+(.+)\s*$');
+
+  @override
+  bool canParse(md.BlockParser parser) {
+    return pattern.hasMatch(parser.current.content);
+  }
+
+  /// Parses the block of text for the given spoiler. This will fetch the title and the body of the spoiler.
+  @override
+  md.Node parse(md.BlockParser parser) {
+    final Match? match = pattern.firstMatch(parser.current.content);
+    final String? title = match?.group(1)?.trim();
+
+    parser.advance(); // Move to the next line
+
+    final List<String> body = [];
+
+    // Accumulate lines of the body until the closing :::
+    while (!parser.isDone) {
+      if (_spoilerBlockEnd.hasMatch(parser.current.content)) {
+        parser.advance();
+        break;
+      }
+      body.add(parser.current.content);
+      parser.advance();
+    }
+
+    // Create a custom Node which will be used to render the spoiler in [SpoilerElementBuilder]
+    final md.Node spoiler = md.Element('p', [
```

Comment:
@micahmo I encountered the same issue when I tried to add a custom tag in here (as you mentioned in https://github.com/thunder-app/thunder/issues/304#issuecomment-1660486274

The workaround here is to use a known element (e.g., `p` in this case), and then supply the custom element as the child of this (e.g., `spoiler`)

Reply:
Wow great find! I'm so glad you tried it out!